scripts/zones/Temenos/bcnms/central_temenos_2nd_floor.lua

```diff
@@ -2,43 +2,39 @@
 -- Area: Temenos
 -- Name: Central Temenos 2nd Floor
 -----------------------------------
-require("scripts/globals/limbus");
+require("scripts/globals/limbus")
 require("scripts/globals/battlefield")
-require("scripts/globals/keyitems");
+require("scripts/globals/keyitems")
+
+function onBattlefieldInitialise(battlefield)
+    battlefield:setLocalVar("loot", 1)
+    HideTemenosDoor(Central_Temenos_2nd_Floor)
+    HideArmouryCrates(Central_Temenos_2nd_Floor, TEMENOS)
+    if GetMobByID(16929039):isSpawned() then DespawnMob(16929039) end
+    if GetMobByID(16929040):isSpawned() then DespawnMob(16929040) end
+    if GetMobByID(16929041):isSpawned() then DespawnMob(16929041) end
+    if GetMobByID(16929042):isSpawned() then DespawnMob(16929042) end
+    if GetMobByID(16929043):isSpawned() then DespawnMob(16929043) end
+    if GetMobByID(16929044):isSpawned() then DespawnMob(16929044) end
```

Comment:
I can has for loop? Also why do we do this onInit instead of when the battlefield is over with? (I realize the old implementation basically did the same thing. To be fair I hated the old implementation. If it has to be this way just state the reasoning.)

Reply:
p.s. IDs.lua is where we should stuff NPC or MOB IDs whenever possible, and then just call them into other scripts by their then human readable variable names.

Reply:
I plan on cleaning this all up more after I look at Apollyon. This commit is basically just hooking it up with the new bcnm stuff to get it working. I put that stuff in init because I was having issues where a party member could reset the doors by zoning and you'd get stuck. The messageSpecial thing just worked and the only place the function is being used is Limbus. I admittedly don't really know the difference. The IDs is something I wanna change soon but just wanted to get this out of the way first.

Reply:
Also, these are being despawned because they're the avatar mimic mobs. They get spawned on init because I needed to put them into bcnm_battlefield to be able to attack them when they pop but they need to be despawned until they're popped by the player.

Reply:
Not sure I understand. ...Wait, are these those a weird set where its using both an NPC -and- a mob for the same object (single object), rather than depopping the NPC and Spawning a mob in the same place (2 separate objects, same location)? I think I remember seeing that before.

Normally we'd just not spawn the mob until needed in the first place - it wouldn't be up before hand unless it wasn't added to the battlefield proper which would cause it to still be up after the battlefield was over.

Reply:
I don't think so. The treasure chest despawns and the avatar pops correctly, but unless the avatar is in bcnm_battlefield it doesn't get added to the moblist for the battlefield and you're unable to attack it. A side effect of adding it to bcnm_battlefield is that it pops on init, so it needs to be despawned and then it can be popped and attacked normally.

Reply:
> A side effect of adding it to bcnm_battlefield is that it pops on init

That not how that is supposed to work: you can have things be part of the battlefield without them being immediately spawned. We had that working prior to the rewrite, but limbus was never updated.

Reply:
```c++
enum BATTLEFIELDMOBCONDITION : uint8
{
    CONDITION_NONE = 0x00,
    CONDITION_SPAWNED_AT_START = 0x01,
    CONDITION_WIN_REQUIREMENT = 0x02
};
```
If you have them in the table with a value of 1, they are spawned with the bcnm (or a value of 3 because that is spawned + required kill for win condition). If they are in the table but the value was 0 or 2, they should not auto spawn.